### PR TITLE
REST endpoints automatic documentation

### PR DESCRIPTION
**Pull request by fabtagliaferro**
Integration of https://github.com/swaggo/swag to automatically generate RESTful API documentation with OpenApi / Swagger 2.0

## Checklist
- [x] Targeted PR against correct branch
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For Admin Use:
- [ ] Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- [ ] Reviewers Assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]"

### CODE REVIEW

**Comment by marcotradenet** [comment]
I think it's better to add the `/swagger` folder to the `.gitignore`: the swagger documents will be generated during the build and any result of the build should be out of the code base.

**Comment by fabtagliaferro** [comment]
> I think it's better to add the `/swagger` folder to the `.gitignore`: the swagger documents will be generated during the build and any result of the build should be out of the code base.

Totally agree with you!

File: x/documents/client/rest/query.go [hunk 1]
```diff
@@ -51,6 +51,15 @@ func RegisterRoutes(cliCtx context.CLIContext, r *mux.Router) {
...
 // ----------------------------------
 
+// @Summary Get the sent documents
+// @Description This endpoint returns the sent documents, along with the height at which the resource was queried at
+// @ID getSentDocumentsHandler
+// @Produce json
+// @Param address path string true "Address of the user"
+// @Success 200 {object} x.JSONResult{result=[]types.Document}
+// @Failure 404
+// @Router /documents/{address}/sent [get]
```
Comment (marcotradenet) [inline x/documents/client/rest/query.go:61]: Rest endpoint is `docs` not `documents`. 
For example: if you call

http://localhost:1317/documents/did:com:12p24st9asf394jv04e8sxrl9c384jjqwejv0gf/sent

return error `404`

File: x/documents/client/rest/query.go [hunk 2]
```diff
@@ -67,6 +76,15 @@ func getSentDocumentsHandler(cliCtx context.CLIContext) http.HandlerFunc {
...
 }
 
+// @Summary Get the received documents
+// @Description This endpoint returns the received documents, along with the height at which the resource was queried at
+// @ID getReceivedDocumentsHandler
+// @Produce json
+// @Param address path string true "Address of the user"
+// @Success 200 {object} x.JSONResult{result=[]types.Document}
+// @Failure 404
+// @Router /documents/{address}/received [get]
```
Comment (marcotradenet) [inline x/documents/client/rest/query.go:86]: Rest endpoint is `docs` not `documents`.

File: x/documents/client/rest/query.go [hunk 3]
```diff
@@ -123,6 +159,14 @@ func getReceivedDocumentsReceiptsHandler(cliCtx context.CLIContext) http.Handler
 // --- Document metadata schemes
 // ----------------------------------
 
+// @Summary Get the metadata schemes
+// @Description This endpoint returns the supported metadata schemes, along with the height at which the resource was queried at
+// @ID getSupportedMetadataSchemesHandler
+// @Produce json
+// @Success 200 {object} x.JSONResult{result=[]types.MetadataSchema}
+// @Failure 404
+// @Router /documents/metadataSchemes [get]
```
Comment (marcotradenet) [inline x/documents/client/rest/query.go:168]: Rest endpoint is `docs` not `documents`.

File: x/documents/client/rest/query.go [hunk 4]
```diff
@@ -140,6 +184,14 @@ func getSupportedMetadataSchemesHandler(cliCtx context.CLIContext) http.HandlerF
 // --- Document metadata schemes proposers
 // -----------------------------------------
 
+// @Summary Get the metadata proposers
+// @Description This endpoint returns the trusted metadata proposers, along with the height at which the resource was queried at
+// @ID getTrustedMetadataSchemesProposersHandler
+// @Produce json
+// @Success 200 {object} x.JSONResult{result=[]string}
+// @Failure 404
+// @Router /documents/metadataSchemes/proposers [get]
```
Comment (marcotradenet) [inline x/documents/client/rest/query.go:193]: Rest endpoint is `docs` not `documents`.

**Review by marcotradenet — CHANGES_REQUESTED** [review]

File: x/swagger.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+package x
+
+import (
+	_ "github.com/cosmos/cosmos-sdk/types"
+)
+
+// @title CommercioNetwork API
+// @description Swagger API for CommercioNetwork
+// @contact.name Gianguido Sorà
```
Comment (gsora) [inline x/swagger.go:11]: Maybe we should use something more generic for `@contact.name` and `@contact.email`.

@marcotradenet what could we use?

**Review by gsora — CHANGES_REQUESTED** [review]

File: x/swagger.go (hunk 5 above)
Comment (marcotradenet) [reply x/swagger.go:11]: I think the correct email should be developers@commercio.network

**Review by gsora — APPROVED** [review]
LFTM

**Review by marcotradenet — APPROVED** [review]

**Pull request merged**